engine/src/main/java/org/terasology/logic/characters/VisualCharacterSystem.java

```diff
@@ -0,0 +1,128 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.characters;
+
+import org.terasology.assets.management.AssetManager;
+import org.terasology.engine.modes.loadProcesses.AwaitedLocalCharacterSpawnEvent;
+import org.terasology.entitySystem.entity.EntityBuilder;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnActivatedComponent;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.location.Location;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.logic.characters.events.CreateVisualCharacterEvent;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.network.ClientComponent;
+import org.terasology.network.ColorComponent;
+import org.terasology.registry.In;
+import org.terasology.rendering.logic.SkeletalMeshComponent;
+import org.terasology.rendering.nui.Color;
+
+import java.util.Optional;
+
+/**
+ * This system is responsible for sending a {@link CreateVisualCharacterEvent} according to how it is specified in
+ * {@link VisualCharacterComponent}. It also provides a default handling for {@link CreateVisualCharacterEvent} that
+ * creates a floating cube.
+ */
+@RegisterSystem(RegisterMode.CLIENT)
+public class VisualCharacterSystem extends BaseComponentSystem {
+    @In
+    LocalPlayer localPlayer;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private AssetManager assetManager;
+
+    private boolean awaitedLocalCharacterSpawn = false;
+
+    @ReceiveEvent(components = {VisualCharacterComponent.class})
+    public void onActivatedVisualCharacter(OnActivatedComponent event, EntityRef entity) {
+        if (!awaitedLocalCharacterSpawn) {
+            /*
+             * Before character has spawned localPlayer is not properly initialized
+             * and can not be used to test if character is visible.
+             */
+            return;
+        }
+        sendCreateVisualCharacterEventIfNotOwnCharacter(entity);
+    }
+
+    void sendCreateVisualCharacterEventIfNotOwnCharacter(EntityRef entity) {
```

Comment:
I wonder,  if in some future land we have a 3rd person camera,  we may likely want to have this event still sent. Maybe with a flag that it is for the local player,  and then let the mesh adding system do the logic whether or not it wants to add it for the local player's character?

Alternatively,  cancel the event if the player is in first person mode?  (not sure if this would be a good idea)